dp/generate_climos.py

```diff
@@ -39,17 +39,96 @@
 cdo = Cdo()
 
 
+def input_check(filepath, climo):
+    '''
+    This function runs general checks on the given input arguments filepath and climo.
+    First, it checks if the given input file path is a vaild NetCDF file. It returns
+    NetCDF dataset of the input file. Ohterwise, it halts the program. Next, it checks
+    if the input NetCDF dataset has a set of standard climo periods by using climo_periods
+    property. It returns the periods if found, outherwise it halts the program.
+    '''
+    logger.info("")
+    logger.info("Processing: {}".format(filepath))
+
+    try:
+        input_file = CFDataset(filepath)
+    except OSError as e:
+        logger.info("{}: {}".format(e.__class__.__name__, e))
+        return
+
+    periods = input_file.climo_periods.keys() & climo
+    logger.info("climo_periods: {}".format(periods))
+    if len(periods) == 0:
+        logger.critical(f"{input_file.filepath()} has no 'climo_periods' found")
+        return
+
+    return input_file, periods
+
+
+def dry_run_handler(filepath, climo):
+    '''
+    This function handles dry-run operation of generate_climos. The purpose of this function
+    is to check variables and attrbutes to be used for generate_climos. dry-run will not 
+    produce any output files.
+    '''
+    input_file, periods = input_check(filepath, climo)
+
+    for attr in ['project', 'institution', 'model', 'emissions', 'run']:
+        try:
+            logger.info(
+                "{}: {}".format(attr, getattr(input_file.metadata, attr))
+            )
+        except Exception as e:
+            logger.info("{}: {}: {}".format(attr, e.__class__.__name__, e))
+    logger.info(
+        "dependent_varnames: {}".format(input_file.dependent_varnames())
+    )
+    for attr in ['time_resolution', 'is_multi_year_mean']:
+        logger.info("{}: {}".format(attr, getattr(input_file, attr)))
+
+
+def generate_climos(
+    filepath,
+    outdir,
+    operation,
+    climo=standard_climo_periods().keys(),
+    convert_longitudes=True,
+    split_vars=True,
+    split_intervals=True,
+    resolutions={"yearly", "seasonal", "monthly"},
+):
+    '''
+    This function runs general generate_climos operation. The main purpose of this function
+    is to call create_climo_files.
+    '''
+    input_file, periods = input_check(filepath, climo)
+
+    for period in periods:
+        t_range = input_file.climo_periods[period]
+        create_climo_files(
+            period,
+            outdir,
+            input_file,
+            operation,
+            *t_range,
+            convert_longitudes=convert_longitudes,
+            split_vars=split_vars,
+            split_intervals=split_intervals,
+            output_resolutions=resolutions,
+        )
```

Comment:
This is very nice. The two methods `dry_run_handler` and `generate_climos` are short, simple, and clear.